Gordon360/Services/RecIM/TeamService.cs

```diff
@@ -0,0 +1,176 @@
+                                && opmt.TeamID != teamID)
+                                .Score)
+                                .Count(),
+                        }),
+                }).FirstOrDefault();
+            return team;
+        }
+
+        public async Task<int> PostTeamAsync(TeamUploadViewModel t, string username)
+        {
+            var participantID = _context.ACCOUNT
+                                .FirstOrDefault(a => a.AD_Username == username)
+                                .account_id;
+
+            var team = new Team
+            {
+                Name = t.Name,
+                StatusID = t.StatusID,
+                ActivityID = t.ActivityID,
+                Private = t.Private,
+                Logo = t.Logo,
+            };
+            await _context.Team.AddAsync(team);
+            await _context.SaveChangesAsync();
+
+            await AddUserToTeamAsync(team.ID, username);
+            await UpdateParticipantRoleAsync(team.ID, participantID, 5);
+
+            return team.ID;
+        }
+
+        public async Task UpdateParticipantRoleAsync(int teamID, int participantID, int participantRoleID)
+        {
+            var participantTeam = _context.ParticipantTeam.FirstOrDefault(pt => pt.ParticipantID == participantID && pt.TeamID == teamID);
+            participantTeam.RoleType = participantRoleID;
+
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task<int> UpdateTeamAsync(int teamID, TeamPatchViewModel update)
+        {
+            var t = await _context.Team.FindAsync(teamID);
+            t.Name = update.Name ?? t.Name;
+            t.StatusID = update.StatusID ?? t.StatusID;
+            t.Private = (bool)update.Private;
+            t.Logo = update.Logo ?? t.Logo;
+
+            await _context.SaveChangesAsync();
+
+            return teamID;
+        }
+        public async Task AddUserToTeamAsync(int teamID, string username)
+        {
+            var participantID = _context.ACCOUNT
+                                .FirstOrDefault(a => a.AD_Username == username)
+                                .account_id;
+
+            var participantTeam = new ParticipantTeam
+            {
+                TeamID = teamID,
+                ParticipantID = participantID,
+                SignDate = DateTime.Now,
+                RoleType = 3, //default: 3 -> member
+            };
+            await _context.ParticipantTeam.AddAsync(participantTeam);
+
+            await _context.SaveChangesAsync();
+        }
+
+        public bool IsTeamCaptain(int teamID, string username)
+        {
+            var participantID = Int32.Parse(_context.ACCOUNT
+                                .FirstOrDefault(a => a.AD_Username == username)
+                                .gordon_id);
+            return _context.ParticipantTeam.FirstOrDefault(t => t.TeamID == teamID && t.ParticipantID == participantID).RoleType == 5;
```

Comment:
I would use a .Any(...) LINQ method here and put it all inside the lambda function:
```suggestion
            return _context.ParticipantTeam.Any(t =>
                        t.TeamID == teamID
                        && t.ParticipantID == participantID
                        && t.RoleType == 5
            );
```

Reply:
I forgot to specify, but .Any() will return `true` if any of the lambda evaluations inside of it result to `true`.